src/cgnslib.c

```diff
@@ -5039,12 +5039,13 @@ int cg_poly_elements_general_read(int file_number, int B, int Z, int S,
     else if (m_type == CGNS_ENUMV(LongInteger)) {
         cglong_t* tmp_connect_offset = (cglong_t*)connect_offset;
         cglong_t offset = tmp_connect_offset[0];
-        cglong_t size = tmp_connect_offset[end - start + 1] - offset;
-        if (size < 0) return CG_ERROR;
+        cglong_t size_long = tmp_connect_offset[end - start + 1] - offset;
+        if (size_long < 1) return CG_ERROR;
+        size = (cgsize_t)size_long;
```

Comment:
Shouldn't we check if this will fit before assigning it?

Reply:
Yes, it would be safer to add a check to limit overflow.
Right now, since "size_long" is the result of a difference the problem would happen only if the user try to read a big array at once and not partially.